### ISSUE TITLE
IPython console error

### ISSUE THREAD

**Issue by Yoyo8274** [spyder-ide/spyder#17263]
### What steps will reproduce the problem?

While changing 'Tools' -> 'Preference' -> 'IPython console' -> 'Graphics' -> 'Graphics Backend' from Tkinter to Inline prblem is generating. I am using Spyder5(python 3.9) with Anaconda3 in Windows 10. My hardware specification is 6 Gb ram and intel pentium processor. 

### Traceback
```python-traceback
Traceback (most recent call last):
  File "C:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\preferences\widgets\configdialog.py", line 127, in accept
    configpage.apply_changes()
  File "C:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\preferences\api.py", line 110, in apply_changes
    self.apply_callback()
  File "C:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\preferences\api.py", line 142, in <lambda>
    self._apply_settings_tabs(self.changed_options))
  File "C:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\preferences\api.py", line 170, in _apply_settings_tabs
    self.apply_settings(options)
  File "C:\ProgramData\Anaconda3\lib\site-packages\spyder\api\preferences.py", line 104, in wrapper
    func(opts)
  File "C:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\ipythonconsole\plugin.py", line 452, in apply_plugin_settings
    client_backend_not_inline = [
  File "C:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\ipythonconsole\plugin.py", line 453, in <listcomp>
    client.shellwidget.get_matplotlib_backend() != inline_backend
  File "C:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\ipythonconsole\widgets\shell.py", line 351, in get_matplotlib_backend
    return self.call_kernel(
  File "C:\ProgramData\Anaconda3\lib\site-packages\spyder_kernels\comms\commbase.py", line 554, in __call__
    return self._comms_wrapper._get_call_return_value(
  File "C:\ProgramData\Anaconda3\lib\site-packages\spyder\plugins\ipythonconsole\comms\kernelcomm.py", line 212, in _get_call_return_value
    raise CommError("Cannot block on a disconnected comm")
spyder_kernels.comms.commbase.CommError: Cannot block on a disconnected comm
```

## Versions

* Spyder version: 5.1.5 
* Python version: 3.9.7
* Qt version: 5.9.7
* PyQt5 version: 5.9.2
* Operating System: Windows 10

### Dependencies

```

# Mandatory:
atomicwrites >=1.2.0          :  1.4.0 (OK)
chardet >=2.0.0               :  4.0.0 (OK)
cloudpickle >=0.5.0           :  2.0.0 (OK)
cookiecutter >=1.6.0          :  1.7.2 (OK)
diff_match_patch >=20181111   :  20200713 (OK)
intervaltree >=3.0.2          :  3.1.0 (OK)
IPython >=7.6.0               :  7.29.0 (OK)
jedi >=0.17.2;<0.19.0         :  0.18.0 (OK)
jsonschema >=3.2.0            :  3.2.0 (OK)
keyring >=17.0.0              :  23.1.0 (OK)
nbconvert >=4.0               :  6.1.0 (OK)
numpydoc >=0.6.0              :  1.1.0 (OK)
paramiko >=2.4.0              :  2.7.2 (OK)
parso >=0.7.0;<0.9.0          :  0.8.2 (OK)
pexpect >=4.4.0               :  4.8.0 (OK)
pickleshare >=0.4             :  0.7.5 (OK)
psutil >=5.3                  :  5.8.0 (OK)
pygments >=2.0                :  2.10.0 (OK)
pylint >=2.5.0;<2.10.0        :  2.9.6 (OK)
pyls_spyder >=0.4.0           :  0.4.0 (OK)
pylsp >=1.2.2;<1.3.0          :  1.2.4 (OK)
pylsp_black >=1.0.0           :  None (OK)
qdarkstyle =3.0.2             :  3.0.2 (OK)
qstylizer >=0.1.10            :  0.1.10 (OK)
qtawesome >=1.0.2             :  1.0.2 (OK)
qtconsole >=5.1.0             :  5.1.1 (OK)
qtpy >=1.5.0                  :  1.10.0 (OK)
rtree >=0.9.7                 :  0.9.7 (OK)
setuptools >=49.6.0           :  58.0.4 (OK)
sphinx >=0.6.6                :  4.2.0 (OK)
spyder_kernels >=2.1.1;<2.2.0 :  2.1.3 (OK)
textdistance >=4.2.0          :  4.2.1 (OK)
three_merge >=0.1.1           :  0.1.1 (OK)
watchdog >=0.10.3             :  2.1.3 (OK)
zmq >=17                      :  22.2.1 (OK)

# Optional:
cython >=0.21                 :  0.29.24 (OK)
matplotlib >=2.0.0            :  3.4.3 (OK)
numpy >=1.7                   :  1.19.5 (OK)
pandas >=1.1.1                :  1.4.0 (OK)
scipy >=0.17.0                :  1.7.3 (OK)
sympy >=0.7.3                 :  1.9 (OK)
```

**Comment by Ajax-Light** [comment]
Hey, I've done the steps which you've mentioned:
"While changing 'Tools' -> 'Preference' -> 'IPython console' -> 'Graphics' -> 'Graphics Backend' from Tkinter to Inline"
and I'm unable to reproduce the error.
The Graphics Backend is set to Inline by default. Could you try resetting Spyder to its factory defaults and/or Reinstalling it and see if the problem persists?

**Comment by Ajax-Light** [comment]
Looking at kernelcomm.py, seems like a runtime error and is unfortunately hard to debug :(
Looks like your python kernel got disconnect from the console for whatever reason.
Try changing the settings one more time and see exception is thrown again.

**Comment by ccordoba12** [comment]
Hey @<>, thanks for reporting. This error is already fixed in our latest version (5.2.2).

Since it's still not available in Anaconda, you have two options:

1. Use our Windows installer, which you can find [here](https://github.com/spyder-ide/spyder/releases).
2. Create a new environment with conda-forge packages. For that, please close Spyder, open the Anaconda Prompt and run the following commands there:

        conda create -n spyder-cf -c conda-forge spyder
        conda activate spyder-cf
        spyder

After that, you can connect Spyder to your base environment by following these instructions:

http://docs.spyder-ide.org/current/faq.html#using-existing-environment

**Comment by ccordoba12** [comment]
By the way, thanks for your help @Ajax-Light!
